p2p/src/swarm/mod.rs

```diff
@@ -14,46 +14,33 @@
 // limitations under the License.
 //
 // Author(s): A. Altonen
+
+//! Peer manager
+//!
+//! TODO
+//!
+//!
+
+#![allow(rustdoc::private_intra_doc_links)]
 use crate::{
-    error::{FatalError, P2pError, PeerError, ProtocolError},
+    error::{P2pError, PeerError, ProtocolError},
     event,
     net::{self, ConnectivityService, NetworkingService},
 };
-use common::chain::ChainConfig;
+use common::{chain::ChainConfig, primitives::version};
 use futures::FutureExt;
 use logging::log;
-use std::{
-    collections::{HashMap, HashSet},
-    fmt::Debug,
-    str::FromStr,
-    sync::Arc,
-};
-use tokio::sync::mpsc;
+use std::{collections::HashMap, fmt::Debug, str::FromStr, sync::Arc, time::Duration};
+use tokio::sync::{mpsc, oneshot};
+use utils::ensure;
 
-const MAX_ACTIVE_CONNECTIONS: usize = 32;
+pub mod peerdb;
 
-// TODO: store active address
-// TODO: store other discovered addresses
-#[derive(Debug)]
-struct PeerContext<T>
-where
-    T: NetworkingService,
-{
-    _info: net::types::PeerInfo<T>,
-}
+/// Maximum number of connections the [`PeerManager`] is allowed to have open
+const MAX_ACTIVE_CONNECTIONS: usize = 128;
```

Comment:
I may have mentioned this before, but this should be a command line argument. Not necessarily right now to be done, but a TODO wouldn't be bad.

Reply:
Yeah it's on the TODO list but not with a high priority. I'll add configs to P2P at some point